### PR TITLE
dag/expr: implement flag DividedByZeroAsWarning

### PR DESCRIPTION
**Pull request by AndreMouche**
@hicqu @breeswish @XuHuaiyu PTAL

### CODE REVIEW

File: src/coprocessor/dag/expr/ctx.rs [hunk 1]
```diff
@@ -28,17 +28,24 @@ pub const FLAG_IGNORE_TRUNCATE: u64 = 1;
 /// This flag only matters if `FLAG_IGNORE_TRUNCATE` is not set, in strict sql mode, truncate error
 /// should be returned as error, in non-strict sql mode, truncate error should be saved as warning.
 pub const FLAG_TRUNCATE_AS_WARNING: u64 = 1 << 1;
-
 // `FLAG_PAD_CHAR_TO_FULL_LENGTH` indicates if sql_mode 'PAD_CHAR_TO_FULL_LENGTH' is set.
 pub const FLAG_PAD_CHAR_TO_FULL_LENGTH: u64 = 1 << 2;
-
-// `FLAG_IN_SELECT_STMT` indicates if this is a SELECT statement.
+/// FlagInInsertStmt indicates if this is a INSERT statement.
```
Comment (hicqu) [inline src/coprocessor/dag/expr/ctx.rs]: Why not `FLAG_IN_INSERT_STMT`?

**Comment by hicqu** [comment]
rest LGTM.

**Comment by XuHuaiyu** [comment]
LGTM

File: Cargo.toml [hunk 2]
```diff
@@ -95,6 +95,7 @@ git = "https://github.com/pingcap/kvproto.git"
 
 [dependencies.tipb]
 git = "https://github.com/pingcap/tipb.git"
+branch = "sqlmode"
```
Comment (zz-jason) [inline Cargo.toml]: can be removed

**Pull request merged**